cmd/gardenlet/app/bootstrap.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener/cmd/utils"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	"github.com/gardener/gardener/pkg/gardenlet/bootstrap"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	certificatesv1beta1client "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/keyutil"
+	componentbaseconfig "k8s.io/component-base/config"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+func bootstrapKubeconfig(
+	ctx context.Context,
+	logger *logrus.Logger,
+	gardenClientConnection config.GardenClientConnection,
+	seedClientConnection componentbaseconfig.ClientConnectionConfiguration,
+	seedConfig *config.SeedConfig,
+) ([]byte, error) {
+	if gardenClientConnection.KubeconfigSecret == nil {
```

Comment:
Should this check be ?
```suggestion
	if gardenClientConnection.KubeconfigSecret == nil && gardenClientConnection.BootstrapKubeconfig == nil  {
```

Reply:
no, this is done further down in the function, see line 65